ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py

```diff
@@ -70,10 +70,302 @@ def _check_query(query):
+    assert ta_gt.element_shape == ta.element_shape
+
+
+@given(l_kwargs=_helper_random_tensorarray())
+def test_tensorarray_flow(
+    l_kwargs,
+    backend_fw,
+):
+    ta_gt, ta = _helper_init_tensorarray(backend_fw, l_kwargs)
+    assert ta_gt.flow == ta.flow
+
+
+@given(l_kwargs=_helper_random_tensorarray(fn="gather"))
+def test_tensorarray_gather(
+    l_kwargs,
+    backend_fw,
+):
+    ta_gt, ta = _helper_init_tensorarray(backend_fw, l_kwargs[:2])
+    *_, indices = l_kwargs
+    helpers.value_test(
+        ret_np_from_gt_flat=ta_gt.gather(indices).numpy().flatten(),
+        ret_np_flat=np.array(ta.gather(indices)).flatten(),
+        backend=backend_fw,
+    )
+
+
+@given(l_kwargs=_helper_random_tensorarray())
+def test_tensorarray_handle(
+    l_kwargs,
+    backend_fw,
+):
+    ta_gt, ta = _helper_init_tensorarray(backend_fw, l_kwargs)
+    assert ta_gt.handle == ta.handle
+
+
+@given(l_kwargs=_helper_random_tensorarray(fn="scatter"))
+def test_tensorarray_scatter(
+    l_kwargs,
+    backend_fw,
+):
+    id_read, _ = l_kwargs
+    ta_gt, ta = _helper_init_tensorarray(backend_fw, l_kwargs, "scatter")
+    for id, read in id_read:
+        helpers.value_test(
+            ret_np_from_gt_flat=ta_gt.read(id).numpy().flatten(),
+            ret_np_flat=np.array(ta.read(id)).flatten(),
+            backend=backend_fw,
+        )
+
+
+@given(l_kwargs=_helper_random_tensorarray())
+def test_tensorarray_size(
+    l_kwargs,
+    backend_fw,
+):
+    ta_gt, ta = _helper_init_tensorarray(backend_fw, l_kwargs)
+    helpers.value_test(
+        ret_np_from_gt_flat=ta_gt.size().numpy().flatten(),
+        ret_np_flat=np.array(ta.size()).flatten(),
+        backend=backend_fw,
+    )
+
+
+@given(
+    kwargs_v_l=_helper_tensorarray_split(),
+)
+def test_tensorarray_split(kwargs_v_l, backend_fw):
+    ta_gt, ta = _helper_init_tensorarray(backend_fw, kwargs_v_l, "split")
+    for id in range(ta_gt.size()):
+        helpers.value_test(
+            ret_np_from_gt_flat=ta_gt.read(id).numpy().flatten(),
+            ret_np_flat=np.array(ta.read(id)).flatten(),
+            backend=backend_fw,
+        )
+
+
+@given(l_kwargs=_helper_random_tensorarray(fn="stack"))
+def test_tensorarray_stack(
+    l_kwargs,
+    backend_fw,
+):
+    ta_gt, ta = _helper_init_tensorarray(backend_fw, l_kwargs)
+    helpers.value_test(
+        ret_np_from_gt_flat=ta_gt.stack().numpy().flatten(),
+        ret_np_flat=np.array(ta.stack()).flatten(),
+        backend=backend_fw,
+    )
+
+
+@given(l_kwargs=_helper_tensorarray_unstack())
+def test_tensorarray_unstack(
+    l_kwargs,
+    backend_fw,
+):
+    ta_gt, ta = _helper_init_tensorarray(backend_fw, l_kwargs, "unstack")
+    helpers.value_test(
+        ret_np_from_gt_flat=ta_gt.stack().numpy().flatten(),
+        ret_np_flat=np.array(ta.stack()).flatten(),
+        backend=backend_fw,
+    )
+
+
+# test for read and write methods
+@given(l_kwargs=_helper_random_tensorarray())
+def test_tensorarray_write_read(
```

Comment:
Actually we need the test names to follow the format "test_[frontend name]_[function/method name]". This is important for our AI developer team. So, the comment you left will suffice. No need to change to "write_read".